### DISCUSSION TITLE
With Spring Data, is it needed to create Collections manually?

### DISCUSSION THREAD

**Discussion by MitchelNijdam-Rockstars** [flamingock/mongock#40] [Q&A] [answered]
Hello, 

I want to use Mongock with Spring Data. Currently, by using the `@Document(collection = "test")` annotation of Spring on the data object, the application automatically creates the collections. However, in the [examples](https://github.com/mongock/mongock-examples/blob/master/mongodb/springboot-quickstart/src/main/java/io/mongock/examples/changelogs/ClientInitializerChangeLog.java) of mongock for Spring I see that `mongoTemplate.createCollection()` is called manually from within the ChangeUnit. Is this really necessary? And what are the main reasons to create collections manually or automatically?

**Comment by osantana85** [comment, marked as answer]
Hello @MitchelNijdam-Rockstars , you are right, there is no reason to create the collection manually, it's more about to show "beforeExecution" feature, that allows you to make "schema" changes without transaction scope. But as said, in this case Spring Data will create the collections automatically for you.

**Reply by dieppa** [reply]
@MitchelNijdam-Rockstars I would also add that many people prefer to not rely on those annotations to create the DDLs, but instead having them in "scripts", so they have more control over it. Specially when it comes to production environment.

But once again, not necessary 😄

**Reply by MitchelNijdam-Rockstars** [reply]
Yes, that is also my experience with for example `flyway`. Thanks for the confirmation!
